lib/runners/processor/checkstyle.rb

```diff
@@ -101,11 +101,11 @@ def checkstyle_args
     def construct_result(xml_root)
       ignored_severities = Array(config_linter[:ignore])
 
-      xml_root.each_element("file") do |file|
+      xml_root.search("file").each do |file|
```

Comment:
[question] `#search` method seems to use XPath, so perhaps it may be slow.
In a simpler case, `#traverse` may be faster, but what do you think?

See also:
- https://nokogiri.org/rdoc/Nokogiri/XML/Node.html#search-instance_method
- https://nokogiri.org/rdoc/Nokogiri/XML/Node.html#traverse-instance_method

Reply:
I'm not sure `#traverse` is faster than `#search` because of recursive in a long xml file. 🤔 

> Yields self and all children to `block` recursively.

How about making the specific XPath query?

```suggestion
      xml_root.search("/checkstyle/file").each do |file|
```

Reply:
No, I suggest not using XPath query as below:

```diff
diff --git a/lib/runners/processor/checkstyle.rb b/lib/runners/processor/checkstyle.rb
index 2b56f465..9e3343fd 100644
--- a/lib/runners/processor/checkstyle.rb
+++ b/lib/runners/processor/checkstyle.rb
@@ -101,11 +101,11 @@ module Runners
     def construct_result(xml_root)
       ignored_severities = Array(config_linter[:ignore])
 
-      xml_root.search("file").each do |file|
+      xml_doc.root.element_children.each do |file|
         file_name = file[:name] or raise "Invalid file: #{file.inspect}"
         path = relative_path file_name
 
-        file.search("error").each do |error|
+        file.element_children.each do |error|
           case error.name
           when "error"
             severity = error[:severity]
```

Beside, the `xml_doc` variable name seems better than `xml_root` to me.

Reply:
Sorry, `#traverse` is not suitable, instead `#element_children` is better.